worker/caasunitprovisioner/deployment_worker.go

```diff
@@ -183,15 +183,25 @@ func (w *deploymentWorker) loop() error {
 			if err != nil && !errors.IsNotFound(err) {
 				return errors.Annotate(err, "cannot get new service details")
 			}
-			err = w.applicationUpdater.UpdateApplicationService(params.UpdateApplicationServiceArg{
-				ApplicationTag: names.NewApplicationTag(w.application).String(),
-				ProviderId:     service.Id,
-				Addresses:      params.FromNetworkAddresses(service.Addresses...),
-			})
-			if err != nil {
+			if err = updateApplicationService(
+				names.NewApplicationTag(w.application), service, w.applicationUpdater,
+			); err != nil {
 				return errors.Trace(err)
 			}
 			serviceUpdated = true
 		}
 	}
 }
+
+func updateApplicationService(appTag names.ApplicationTag, svc *caas.Service, updater ApplicationUpdater) error {
+	if svc.Id == "" {
```

Comment:
without this, an empty doc will be created then assertion failed in later `update` ops because we are using `svc.Id` for unq ID `provider-id`